external_libraries/mpi_python/python_interface.cpp

```diff
@@ -23,20 +38,27 @@ PYBIND11_MODULE(mpipython, m)
     RankFuncType FRank = &PythonMPI::rank;
     SizeFuncType FSize = &PythonMPI::size;
 
-    class_<PythonMPI>(m,"PythonMPI") 
+    // note that for the functions returning a vector the conversion to a python-list is automatically
+    // done by pybind, see https://github.com/pybind/pybind11/blob/master/docs/advanced/cast/stl.rst
+    std::vector<int> (PythonMPI::*gather_int)(PythonMPIComm&, const int, const int) = &PythonMPI::gather;
+    std::vector<double> (PythonMPI::*gather_double)(PythonMPIComm&, const double, const int) = &PythonMPI::gather;
+    std::vector<std::vector<int>> (PythonMPI::*gather_list_int)(PythonMPIComm&, const std::vector<int>&, const int) = &PythonMPI::gather;
+    std::vector<std::vector<double>> (PythonMPI::*gather_list_double)(PythonMPIComm&, const std::vector<double>&, const int) = &PythonMPI::gather;
+
+    py::class_<PythonMPI>(m,"PythonMPI")
     .def_property_readonly("rank",FRank)
     .def_property_readonly("size",FSize)
-    .def("gather",&PythonMPI::gather<double>)
-    .def("gather",&PythonMPI::gather<int>)
-    .def("gather", gather_list)
+    .def("gather", gather_int)
+    .def("gather", gather_double)
+    .def("gather", gather_list_int)
```

Comment:
oh i see "gatherv" was not exposed. Btw, shouldn't we call this GatherV (as in the original call) rather than gather? (that is a question for @jcotela )

Reply:
yep this is GatherV

I can change it if wanted, but this will break existing code

Reply:
I guess now would be the time to break things if I anyway refactor it

afaik only @mpentek and @msandre are using this, the impact could be limited

Reply:
@RiccardoRossi I believe this function was added by @msandre. I think we can change it to gatherv for consistency, but I also think we can do it on a separate PR.